app/common/helpers/collections.py

```diff
@@ -73,11 +72,14 @@ def reference(self) -> str:
 
     @property
     def expression_context(self) -> ExpressionContext:
-        return ExpressionContext(
-            from_submission=immutabledict(
-                {mangle_question_id_for_context(uuid.UUID(k)): v for k, v in self.submission.data.items()}
-            )
-        )
+        submission_data = {
+            question.safe_qid: answer.model_dump()
```

Comment:
I think its mentioned in comments but this will work nicely well all the question types are primitives (root models) and will probably need to be looked at again if they contain multiple data fields